cyipopt/scipy_interface.py

```diff
@@ -127,30 +155,49 @@ def gradient(self, x, **kwargs):
 
     def constraints(self, x):
         con_values = []
-        for fun, args in zip(self._constraint_funs, self._constraint_args):
-            con_values.append(fun(x, *args))
+        for i, (fun, args) in enumerate(
+                zip(self._constraint_funs, self._constraint_args)):
+            if self._constraint_funs_with_jacs[i]:
+                con_values.append(
+                    self.evaluate_con_fun_with_jac(i, fun, x, *args)[0])
+            else:
+                con_values.append(fun(x, *args))
         return np.hstack(con_values)
 
     def jacobian(self, x):
         con_values = []
-        for fun, args in zip(self._constraint_jacs, self._constraint_args):
-            con_values.append(fun(x, *args))
+        for i, (fun, jac, args) in enumerate(
+                zip(self._constraint_funs, self._constraint_jacs, self._constraint_args)):
+            if self._constraint_funs_with_jacs[i]:
+                con_values.append(
+                    self.evaluate_con_fun_with_jac(i, fun, x, *args)[1])
+            else:
+                con_values.append(jac(x, *args))
         return np.vstack(con_values)
 
-    def intermediate(
-            self,
-            alg_mod,
-            iter_count,
-            obj_value,
-            inf_pr,
-            inf_du,
-            mu,
-            d_norm,
-            regularization_size,
-            alpha_du,
-            alpha_pr,
-            ls_trials
-            ):
+    def evaluate_con_fun_with_jac(self, i, con_fun, x, *args):
```

Comment:
Any new public methods should have complete docstrings following the numpydoc format.

Reply:
I can't think of a use case in which the user might need this method, so I'd propose to make it private. Moreover, the `i` argument for the i-th constraint doesn't feel pythonic to me, so another reason to make it private. What do you think?

Reply:
Making it private also need deprecation.

Reply:
Oh, sorry, misunderstood. Yes, make it private.